### PR TITLE
Closes #2043

### LINKED ISSUE #2043: This package destroys basic Meteor components
As soon as I installed this, even Spacebars wasn't recognized. Templates weren't recognized.

I don't know how this Atmosphere package was compiled, but it wasn't done correctly. I packaged a small JS library for my own use and it didn't hose my entire application.

This readout is from the web console in Chrome:
```
Uncaught TypeError: Cannot set property 'guid' of undefined
global-imports.js:6 Uncaught TypeError: Cannot read property 'Materialize' of undefined
template.about.js:2 Uncaught ReferenceError: Template is not defined
template.contact.js:2 Uncaught ReferenceError: Template is not defined
template.home.js:2 Uncaught ReferenceError: Template is not defined
template.master-layout.js:2 Uncaught ReferenceError: Template is not defined
template.nav.js:2 Uncaught ReferenceError: Template is not defined
meteor.js:888 Exception in callback of async function: ReferenceError: Deps is not defined
    at Layout.beginRendering (http://localhost:3000/packages/iron_layout.js?4e65c46acdaf0ce6a8b1a479d7b9d0b5c0902062:288:3)
    at Controller.beginRendering (http://localhost:3000/packages/iron_controller.js?5a7eedaafda7001b9d5cb52322a419bbb758968c:290:23)
    at RouteController._runRoute (http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:470:8)
    at Function.Route.dispatch (http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:854:18)
    at route (http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:711:11)
    at boundNext (http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:424:31)
    at http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:22
    at boundNext (http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:371:14)
    at http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:22
    at dispatch (http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:448:3)
iron_layout.js:95 Uncaught ReferenceError: Spacebars is not defined
```

### PR DESCRIPTION
**Pull request by grsabreu**
You can not define Materialize with window.Materialize because Meteor package export won't be able to export the variable.
Tell me if I have to do some changes before merging :).

### CODE REVIEW

**Comment by grsabreu** [comment]
One more thing, for Meteor developers, is better to add uncompressed assets in packages due two things:
1. Better to debug.
2. Meteor compresses assets automatically for us.

So I take the freedom do replace `bin/materialize.js` by `dist/js/materialize.js`.

**Comment by lampe** [comment]
+1

**Comment by JulianKingman** [comment]
+1

**Comment by nelreina** [comment]
+1

**Comment by Crenshinibon** [comment]
+1

**Comment by serkandurusoy** [comment]
@grsabreu while at it, would you mind also updating `package.js` to support the new `addAsset` directive? Because otherwise, this package won't build.

```javascript
// Check if we are post Meteor@1.2 with which api.addFiles() throws error requiring api.addAssets()
var assetDirective = api.addAssets === undefined ? addFiles : addAssets;

// Add material design icons
api[assetDirective]([
  'dist/font/material-design-icons/Material-Design-Icons.eot',
  'dist/font/material-design-icons/Material-Design-Icons.svg',
  'dist/font/material-design-icons/Material-Design-Icons.ttf',
  'dist/font/material-design-icons/Material-Design-Icons.woff',
  'dist/font/material-design-icons/Material-Design-Icons.woff2'
], 'client');

// Add roboto bold font
api[assetDirective]([
  'dist/font/roboto/Roboto-Bold.ttf',
  'dist/font/roboto/Roboto-Bold.woff',
  'dist/font/roboto/Roboto-Bold.woff2'
], 'client');

// Add roboto light font
api[assetDirective]([
  'dist/font/roboto/Roboto-Light.ttf',
  'dist/font/roboto/Roboto-Light.woff',
  'dist/font/roboto/Roboto-Light.woff2'
], 'client');

// Add roboto medium font
api[assetDirective]([
  'dist/font/roboto/Roboto-Medium.ttf',
  'dist/font/roboto/Roboto-Medium.woff',
  'dist/font/roboto/Roboto-Medium.woff2'
], 'client');

// Add roboto regular font
api[assetDirective]([
  'dist/font/roboto/Roboto-Regular.ttf',
  'dist/font/roboto/Roboto-Regular.woff',
  'dist/font/roboto/Roboto-Regular.woff2'
], 'client');

// Add roboto thin font
api[assetDirective]([
  'dist/font/roboto/Roboto-Thin.ttf',
  'dist/font/roboto/Roboto-Thin.woff',
  'dist/font/roboto/Roboto-Thin.woff2'
], 'client');

// Add css and javascript
api.addFiles([
  'dist/css/materialize.css',
  'dist/js/materialize.js'
 ], 'client');
```

**Comment by grsabreu** [comment]
@serkandurusoy https://github.com/grsabreu/materialize/commit/add48e4d8cbc1884b63eea554d4a99495904a4d6
I did that :)

**Comment by serkandurusoy** [comment]
Hah, I had not realised you've already done it, and even more properly :)

**Comment by dcworldwide** [comment]
:+1

**Comment by bkuri** [comment]
:+1:

**Comment by keyscores** [comment]
+1 @Dogfalo can we merge it?

**Comment by brokenseal** [comment]
:+1:

File: package.js [hunk 1]
```diff
@@ -36,7 +34,23 @@ Package.onUse(function (api) {
...
 });
+
+
+function addAssets(api, assets){
+  if(api.addAssets){
+    api.addAssets(assets, 'client');
+  } else {
+    api.addFiles(assets, 'client', {isAsset: true});
+  }
+}
```
Comment (brokenseal) [inline package.js]: New line? :P

Comment (grsabreu) [inline package.js]: Oh, it's true! I'll add one!

**Comment by dandv** [comment]
Yes, the package should include the *un*minified `.js` file.

**Comment by Dogfalo** [comment]
Thank you everyone for finding out the problem. I do not do much meteor development and so I am grateful for your help.

**Comment by grsabreu** [comment]
Nice one, @Dogfalo!
When the new version will be published to Atmosphere?

**Comment by methodbox** [comment]
Yes, please! I am eager to get this rolled into my app with the fix. Thank you @grsabreu. There are many breaking changes with Meteor 1.2.x, and this is actually a common one.

@Dogfalo, if your packaging anything else for Meteor, this should be a consideration.

**Comment by Dogfalo** [comment]
If you want to grab updates before release you can see what I wrote here to use a github repo as a meteor package #2171

**Comment by PolGuixe** [comment]
+1 for new release to be rolled to Atmosphere.

**Comment by donwojtallo** [comment]
+1

**Pull request merged**